client/state/terms/selectors.js

```diff
@@ -5,6 +5,55 @@ import get from 'lodash/get';
 import values from 'lodash/values';
 
 /**
+ * Internal dependencies
+ */
+import {
+	getSerializedTermsQuery
+} from './utils';
+
+/**
+ * Returns true if currently requesting terms for the taxonomies query, or false
+ * otherwise.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @param  {String}  taxonomy Taxonomy slug
+ * @param  {Object}  query  Taxonomy query object
+ * @return {Boolean}        Whether terms are being requested
+ */
+export function isRequestingSiteTaxonomyTermsForQuery( state, siteId, taxonomy, query ) {
```

Comment:
In fact, we could probably do away with "Site" here too, given that we don't need to make the same distinction as we do in posts (where `getPost` accepts a global ID and `getSitePost` is specific to a site)